library/std/src/io/stdio.rs

```diff
@@ -903,6 +905,44 @@ impl fmt::Debug for StderrLock<'_> {
     }
 }
 
+/// Prints the given `str` and reads a [`String`] from
+/// [standard input](Stdin). The trailing newline is stripped.
+/// Gives an error on EOF (end of file).
+///
+/// # Note
+///
+/// If you require more explicit control over capturing
+/// user input, see the [`Stdin::read_line`] method.
+///
+/// # Examples
+///
+/// ```no_run
+/// #![feature(input)]
+/// use std::io;
+///
+/// fn main() {
+///     let name = io::input("Enter name: ").expect("input failed!");
+///
+///     println!("Your name is {}!", name);
+/// }
```

Comment:
Propagating the error up with `?` is preferred over panicking. I would also be fine with not including the `main` declaration in the example.

```suggestion
/// fn main() -> io::Result<()> {
///     let name = io::input("Enter name: ")?;
///     println!("Your name is {}!", name);
/// 
///     Ok(())
/// }
```